python/tank/util/yaml_cache.py

```diff
@@ -0,0 +1,138 @@
+# Copyright (c) 2015 Shotgun Software Inc.
+# 
+# CONFIDENTIAL AND PROPRIETARY
+# 
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit 
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your 
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights 
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+"""
+Implements a caching mechanism to avoid loading the same yaml file multiple times
+unless it's changed on disk.
+"""
+
+import os
+import copy
+import threading
+
+from tank_vendor import yaml
+from ..errors import TankError
+
+class YamlCache(object):
+    """
+    Main yaml cache class
+    """
+    _instance = None
+
+    @staticmethod
+    def instance():
```

Comment:
I think the more pythonic way is to just have a global variable inside the yaml_cache.py file. Something like this: http://stackoverflow.com/questions/6760685/creating-a-singleton-in-python/6760726#6760726